R/derive_var_ontrtfl.R

```diff
@@ -198,31 +199,32 @@ derive_var_ontrtfl <- function(dataset,
   if (!quo_is_null(filter_pre_timepoint)) {
     dataset <- mutate(
       dataset,
-      new_var_col = if_else(!!filter_pre_timepoint, NA_character_, new_var_col, missing = new_var_col)
+      new_var = if_else(!!filter_pre_timepoint, NA_character_, new_var,
+                            missing = new_var)
     )
   }
 
   if (quo_is_null(ref_end_date)) {
     # Scenario 1: No treatment end date is passed
     dataset <- mutate(
       dataset,
-      new_var_col = if_else(
+      new_var = if_else(
         !is.na(!!ref_start_date) & !is.na(!!start_date) & !!ref_start_date < !!start_date,
         "Y",
-        new_var_col,
-        missing = new_var_col
+        new_var,
+        missing = new_var
       )
     )
   } else {
     # Scenario 2: Treatment end date is passed, window added above
     dataset <- mutate(
       dataset,
-      new_var_col = if_else(
+      new_var = if_else(
```

Comment:
This creates a new variables in the dataset called literally `new_var`. What you want instead is to use the content of the `new_var` variable as the column name. To do so please use `!!new_var :=`. The `!!` makes sure to use the content of the variable not its name and `:=` is required because you cannot have `!!something` on the left hand side of `=`.